### PR TITLE
cln-grpc: do not generate cerficates when no autostart.

### PR DESCRIPTION
**Pull request by tonyaldon**
When we don't set `grpc-port` option, the `cln-grpc` plugin is not started.  So, we shouldn't generate certificates in that case.

### CODE REVIEW

**Review by cdecker — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6755/commits/87fb39a3abf8894bcf12f3ab2ba26260a0f61d27

File: tests/test_cln_rs.py [hunk 1]
```diff
@@ -178,11 +179,16 @@ def test_grpc_generate_certificate(node_factory):
...
     """Ensure that we do not start cln-grpc unless a port is configured.
+    Also check that we do not generate certificates.
     """
     l1 = node_factory.get_node()
 
     wait_for(lambda: [p for p in l1.rpc.plugin('list')['plugins'] if 'cln-grpc' in p['name']] == [])
     assert l1.daemon.is_in_log(r'plugin-cln-grpc: Killing plugin: disabled itself at init')
+    p = Path(l1.daemon.lightning_dir) / TEST_NETWORK
+    files = os.listdir(p)
+    pem_files = [f for f in files if re.match(r".*\.pem$",f)]
```
Comment (vincenzopalazzo) [inline tests/test_cln_rs.py:190]:
```suggestion
    pem_files = [f for f in files if re.match(r".*\.pem$", f)]
```

This cause the following error

```
2023-10-10T09:08:00.8610981Z checking for ANSI C header files... yes
2023-10-10T09:08:00.9901583Z checking for sys/types.h... yes
2023-10-10T09:08:01.1104640Z checking for sys/stat.h... tests/test_cln_rs.py:190:58: E231 missing whitespace after ','
2023-10-10T09:08:01.1687731Z yes
2023-10-10T09:08:01.1774097Z make: *** [Makefile:538: check-python-flake8] Error 1
2023-10-10T09:08:01.1774740Z make: *** Waiting for unfinished jobs....
2023-10-10T09:08:01.3928110Z checking for stdlib.h... yes
2023-10-10T09:08:01.6206722Z checking for string.h... yes
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
I think it is missing just some python-format changes and we are good

**Comment by tonyaldon** [comment]
Thanks @vincenzopalazzo for helping me with my first PRs in CLN and showing me exactly what to change.

I also started to look at the CI logs to find the errors by myself (but maybe better not doing them in the first place).

This is the second time this week I haven't checked Python flake8 before opening a PR.  I'll try not to do that mistake in future.

**Comment by vincenzopalazzo** [comment]
>This is the second time this week I haven't checked Python flake8 before opening a PR. I'll try not to do that mistake in future.

No problem the review are here for this :) I find always Python formatting is a little bit strange :)

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6755/commits/00a60a3608210de7e6dc1c4f73b03634a668f5dd

**Pull request merged**